README.md

```diff
@@ -1,9 +1,205 @@
+```
+
+This enables or disables the provider named `<provider>`. Each provider is
+enabled by default. The `dl` option builds the provider as a loadable library
+and the `PATH` option enables the provider and uses the installation given in
+`PATH`.
+
+### Examples
+
+Consider the following example:
+
+```
+$ ./configure --prefix=/opt/libfabric --disable-sockets && make -j 32 && sudo make install
+```
+This will tell Libfabric to disable the `sockets` provider, and install
+Libfabric in the `/opt/libfabric` tree. All other providers will be enabled if
+possible and all debug features will be disabled.
+
+Alternatively:
+
+```
+$ ./configure --prefix=/opt/libfabric --enable-debug --enable-psm=dl && make -j 32 && sudo make install
+```
+
+This will tell Libfabric to enable the `psm` provider as a loadable library,
+enable all debug code paths, and install Libfabric to the `/opt/libfabric`
+tree. All other providers will be enabled if possible.
+
+
+## Providers
+
+### Sockets
+
+***
+
+The `sockets` provider is a general purpose provider that can be used on any
+system that supports TCP sockets.  The provider is not intended to provide
+performance improvements over regular TCP sockets, but rather to allow
+developers to write, test, and debug application code even on platforms
+that do not have high-performance fabric hardware.  The sockets provider
+supports all libfabric provider requirements and interfaces.
+
+See the `fi_sockets(7)` man page for more details.
+
+### usNIC
+
+***
+
+The `usNIC` provider is designed to run over the Cisco VIC (virtualized NIC)
+hardware on Cisco UCS servers. It utilizes the Cisco usNIC (userspace NIC)
+capabilities of the VIC to enable ultra low latency and other offload
+capabilites on Ethernet networks.
+
+See the `fi_usnic(7)` man page for more details.
+
+#### Dependencies
+
+- The `usNIC` provider depends on having either `libnl1` or `libnl3`. If the
+  provider is being built from source, then the `libnl` header files will need
+  to be provided (e.g. install the "--devel" versions of the package).
+
+#### Configure options
+
+```
+--with-libnl=<directory>
+```
+
+If specified, look for libnl support. If it is not found then the `usNIC`
+provider will not be built. If `<directory>` is specified, then check in the
+directory and check for `libnl3`. If `libnl3` is not found, then check for
+`libnl1`. If no `<directory>` argument is specified, then this option is
+redundant with `--with-usnic` (default).
+
+### GNI
+
+***
+
+The `GNI` provider runs on Cray XC (TM) systems utilizing the user-space
+Generic Network Interface (`uGNI`) which provides low-level access to
+the Aries interconnect.  The Aries interconnect is designed for
+low-latency one-sided messaging and also includes direct hardware
+support for common atomic operations and optimized collectives.
+
+See the `fi_gni(7)` man page for more details.
+
+#### Dependencies
+
+- The `GNI` provider requires `gcc` version 4.9 or higher.
+
```

Comment:
there are more dependencies but @sungeunchoi or I can add after the PR is merged